spec/controllers/comments_controller_spec.rb

```diff
@@ -30,6 +30,18 @@
       expect(response).to redirect_to(proposal.cart)
     end
 
+    it "allows a delegate to comment" do
+      approver = proposal.approvals.first.user
+      delegate = FactoryGirl.create(:user)
+      approver.add_delegate(delegate)
+
+      login_as(delegate)
+
+      expect {
+        post :create, params
+      }.to change{ proposal.comments.count }.from(0).to(1)
```

Comment:
Would you mind adding a check that the comment was "made" by the delegated user?